### DISCUSSION TITLE
IOS自动上传短信验证码

### DISCUSSION THREAD

**Discussion by Jeric-X** [Jeric-X/SyncClipboard#60] [Show and tell]
先导入**手动同步**的快捷指令`Clipboard EX`(在[主页](https://github.com/Jeric-X/SyncClipboard?tab=readme-ov-file#ios)中获取)，然后跟随视频演示操作，小于IOS17时，每次收到验证码时需要手动确认

<details>
<summary>点击展开/折叠视频</summary>

https://github.com/Jeric-X/SyncClipboard/assets/10058586/af2c9ca2-aec8-48fd-8b9d-80db6045e799

</details>

根据 @feranwq 提供的正则表达式，视频中的快捷指令可以优化为图中的样式，注意在`匹配文本`后额外添加了一个`获取匹配文本的组`操作 

<details>
<summary>点击展开/折叠截图</summary>

<img width="603" height="1311" alt="IMG_7018" src="https://github.com/user-attachments/assets/39f22739-b810-4179-a69e-2e300d055856" />

</details>

截图中的正则表达式
```regex
(.*)((代|授权|验证|动态|校验)码|[【\[].*[】\]]|[Cc][Oo][Dd][Ee]|[Vv]erification\s?([Cc]ode)?)\s?(G-|<#>)?([:：\s是为]|[Ii][Ss]){0,3}[\(（\[【{「]?(([0-9\s]{4,6})|([\dA-Za-z]{5,6})(?!([Vv]erification)?([Cc][Oo][Dd][Ee])|:))[」}】\]）\)]?(?=([^0-9a-zA-Z]|$))(.*)
```

修改为此正则表达式后，你也可以将视频内设置`信息包含`的位置留空，使此快捷指令接管所有短信

**Comment by feranwq** [comment]
分享一下我修改后的快捷指令，可以直接在自动化-执行里选择这个快捷指令。
[ios短信自动上传](https://www.icloud.com/shortcuts/978284fdda034256a53dd29d71d753d6)

**Reply by Jeric-X** [reply]
已经不像是人能写出来的正则表达式了

**Reply by feranwq** [reply]
确实，正则来源[smsforwarder]，用下来还是比较稳的，我优化了下验证码后面带空格的匹配。(https://github.com/pppscn/SmsForwarder/wiki/【必读】常见问题#q如何一个正则表达式匹配全部验证码短信)
原版的指令有个VPN的验证码会触发不支持多文件分享，所以我把文件相关的流程移除了。另外iOS的短信过滤不支持正则，干脆把提取验证码加到指令里了，方便添加自动化流程

**Reply by lyx-v** [reply]
(([0-9\s]{4,6}) |([\dA-Za-z]{5,6}) 快捷指令中的这段，{4,6}) | 这里|前面好像多了个空格，会导致4位验证码匹配不到

**Reply by Jeric-X** [reply]
> (([0-9\s]{4,6}) |([\dA-Za-z]{5,6}) 快捷指令中的这段，{4,6}) | 这里|前面好像多了个空格，会导致4位验证码匹配不到

好像真是

**Comment by huaxianyan** [comment]
@Jeric-X @feranwq 
请问一下，更新 iOS 26 后发现这两个方式都无效了，手动上传的快捷指令是没问题的，只有这个验证码自动上传失效了

**Comment by Jeric-X** [comment]
@huaxianyan 刚一升级26我的自动化也失效了，关闭重新打开自动化后又可以自动触发了，你可以尝试关闭再启用或者删除重新添加试一下

**Reply by LemonJerryX** [reply]
关闭再打开这个操作成功了，感谢分享！

**Comment by huaxianyan** [comment]
@Jeric-X 我后来重新搞了下，确实可以，但是有的时候有不行，感觉是 iOS26 的问题，但是又不知道怎么解决

现在退不回去了难受

**Comment by Jeric-X** [comment]
我这也是间歇性失灵，IOS每次更新系统都一堆问题



------------------ 原始邮件 ------------------
发件人: NeKo7inA ***@***.***&gt;
发送时间: 2025年10月1日 15:39
收件人: Jeric-X/SyncClipboard ***@***.***&gt;
抄送: JericX ***@***.***&gt;, Mention ***@***.***&gt;
主题: Re: [Jeric-X/SyncClipboard] IOS自动上传短信验证码 (Discussion #60)

**Comment by huaxianyan** [comment]
@Jeric-X 我在互联网上好像找到原因了，参考 https://discussionschinese.apple.com/thread/256140344?sortBy=rank

目前关掉筛选未知发件人测试了几条都正常工作了，还需要更进一步测试

**Comment by Aobing-code** [comment]
@Jeric-X 3.1.1以后似乎不能用了
